src/reporters/http_sender.js

```diff
@@ -122,6 +122,9 @@ export default class HTTPSender {
       this._logger.error(error);
       SenderUtils.invokeCallback(callback, numSpans, error);
     });
+    req.on('timeout', () => {
+      req.abort();
```

Comment:
What happens to on-error path in this case? Is it still executed? If not, we need to execute the same logic

Reply:
An error event should be emitted following the timeout event (error in this case would be "socket hang up").

Reply:
Hmm... we don't need wait timeout event, because the data has already been sent and the socket will not be reused - so we can abort connection, as soon as we get the response.

Reply:
@yrsh why abort the connection after every send? I'd think it's better to keep it open and reuse for future sends.

Reply:
@yurishkuro because it will not be reused in current case, so it’s better to close connections than to fail with socket limit error.

Reply:
clarification from last project call - "abort connection" above is actually "abort current request",  which makes sense. Without aborting the request the connection is kept open.

Reply:
This will abort the request and destroy the connection.

https://nodejs.org/api/http.html#http_request_abort